Sources/BrowserServicesKit/Email/EmailKeychainManager.swift

```diff
@@ -159,38 +162,45 @@ private extension EmailKeychainManager {
             return
         }
         
-        deleteItem(forField: field)
+        try deleteItem(forField: field)
         try add(data: stringData, forField: field)
     }
     
-    static func add(data: Data, forField field: EmailKeychainField) throws {
-        let query: [String: Any] = [
-            kSecClass as String: kSecClassGenericPassword,
-            kSecAttrSynchronizable as String: false,
-            kSecAttrService as String: field.keyValue,
-            kSecAttrAccessible as String: kSecAttrAccessibleAfterFirstUnlock,
-            kSecValueData as String: data]
+    static func add(data: Data, forField field: EmailKeychainField, useDataProtectionKeychain: Bool = true) throws {
+        let query = [
+            kSecClass: kSecClassGenericPassword,
+            kSecAttrSynchronizable: false,
+            kSecAttrService: field.keyValue,
+            kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlock,
+            kSecValueData: data,
+            kSecUseDataProtectionKeychain: useDataProtectionKeychain] as [String: Any]
         
         let status = SecItemAdd(query as CFDictionary, nil)
         
         if status != errSecSuccess {
-            throw EmailKeychainAccessError.keychainAccessFailure(status)
+            throw EmailKeychainAccessError.keychainSaveFailure(status)
         }
     }
     
-    static func deleteAuthenticationState() {
-        deleteItem(forField: .username)
-        deleteItem(forField: .token)
-        deleteItem(forField: .alias)
-        deleteItem(forField: .cohort)
-        deleteItem(forField: .lastUseDate)
+    static func deleteAuthenticationState() throws {
+        try deleteItem(forField: .username)
+        try deleteItem(forField: .token)
+        try deleteItem(forField: .alias)
+        try deleteItem(forField: .cohort)
+        try deleteItem(forField: .lastUseDate)
     }
     
-    static func deleteItem(forField field: EmailKeychainField) {
+    static func deleteItem(forField field: EmailKeychainField, useDataProtectionKeychain: Bool = true) throws {
         let query: [String: Any] = [
             kSecClass as String: kSecClassGenericPassword,
-            kSecAttrService as String: field.keyValue]
-        SecItemDelete(query as CFDictionary)
+            kSecAttrService as String: field.keyValue,
+            kSecUseDataProtectionKeychain as String: useDataProtectionKeychain]
+        
+        let status = SecItemDelete(query as CFDictionary)
+        
+        if status != errSecSuccess && status != errSecItemNotFound {
```

Comment:
This deliberately ignores cases where you try to delete an item that doesn't exist

Reply:
No strong opinion here, but maybe add a pixel if this flow of deleting inexistent items is ever triggered?

Reply:
I looked into that, but we have logic in the iOS app delegate to remove waitlist state, so this will get tripped legitimately: https://github.com/duckduckgo/iOS/blob/develop/DuckDuckGo/AppDelegate.swift#L417